feat(apple): Add docs for screenshot masking options (#15658)
getsentry/sentry-docs: docs/platforms/apple/common/enriching-events/screenshots/index.mdx

In this example we want to show the message, but not the user name.

```swift
@Binding var user: String

var body: some View {
  VStack {
    Text("Hello")
      .sentryReplayUnmask()
    Text("\(user)")
  }
}
```

In this example, we need to unmask the VStack because its background element will be masked by default.
To hide the username, we need to mask it.

```swift
@Binding var user: String

var body: some View {
  VStack {
    Text("Hello")
    Text("\(user)")
      .sentryReplayMask()
  }
  .background(.blue)
  .sentryReplayUnmask()
}
```

## Debugging Screenshot Masking

To see how elements are being masked, enable the masking preview from anywhere in your app. It will display an overlay on top of the masked elements. This works on the simulator and on device, as well as within Xcode Preview.

```swift
SentrySDK.replay.showMaskPreview()
```

By default, the overlay will be opaque. To configure the opacity, pass the desired opacity as a parameter:

```swift
SentrySDK.replay.showMaskPreview(0.5) // 0.5 opacity to render the preview semi-transparent
```

> Make sure not accidentally include this in your release build by e.g. wrapping it in a `#if DEBUG` block.

Reviewer: ```suggestion
Make sure not to accidentally include this in your release build—for example, by wrapping it in a `#if DEBUG` block.
```

This view modifier works on the simulator and on device, as well as within Xcode Preview. Therefore we recommend to apply the modifier only in your preview code, to ensure proper masking without affecting the final release build.

Note that when you apply this modifier to a view, it will show the masking preview for the entire window containing that view, not just the view itself.

```swift
struct ContentView_Previews: PreviewProvider {
  static var previews: some View {
    ContentView()
      .sentryReplayPreviewMask()
  }
}
```

## Rendering Options

Screenshot rendering can be configured to improve performance. These options control how views are rendered when capturing screenshots.

### Enable View Renderer V2

Starting with SDK v8.50.0, the up-to-5x-faster view renderer V2 is used by default, reducing the impact of screenshot capture on the main thread and potential frame drops. This reduces the amount of time it takes to render the screenshot on the main thread, reducing interruptions and visual lag. Benchmarks have shown significant improvement of up to 4-5x faster rendering (reducing ~160ms to ~36ms) on older devices.

> While we recommend the new view renderer, if you are experiencing issues, you can opt out of using it:

Reviewer 2: `l`: It might not be new anymore at some point.
```suggestion
While we recommend the view renderer V2, if you are experiencing issues, you can opt out of using it:
```
